airflow/models/baseoperator.py

```diff
@@ -896,11 +896,11 @@ def _set_relatives(self, task_or_task_list, upstream=False):
             if dag and not task.has_dag():
                 task.dag = dag
             if upstream:
-                task.add_only_new(task._downstream_task_ids, self.task_id)
+                task.add_only_new(task.get_direct_relative_ids(upstream=False), self.task_id)
                 self.add_only_new(self._upstream_task_ids, task.task_id)
             else:
                 self.add_only_new(self._downstream_task_ids, task.task_id)
```

Comment:
Actually I'm thinking should we make the same change here (for the two lines `self.add_only_new(self._upstream_task_ids, task.task_id)` & `self.add_only_new(self._downstream_task_ids, task.task_id)`) so the lines becomes consistent visually, even though they're the same under the hood.

Let's have other committers' inputs here as well.

Reply:
IMO, both two ways are acceptable, but `self._private_member` is better way to entry private member.

Reply:
Yeah I think it's ok to use private member in this case